### PR TITLE
[python] added f-string to python-package/lightgbm/callback.py

### PR DESCRIPTION
**Pull request by NovusEdge**

### CODE REVIEW

**Comment by NovusEdge** [comment]
#4136

**Review by jameslamb — CHANGES_REQUESTED** [review]
Thanks for this! Please see the linter failures at https://github.com/microsoft/LightGBM/pull/4142/checks?check_run_id=2229486239.

I think you will find the development cycle is faster if you run the linters locally while you test fixes. Please see https://github.com/microsoft/LightGBM/blob/841943f26b9440be01d685060ae1289056623242/.ci/test.sh#L60-L63, and `@` me if you need any help.

File: python-package/lightgbm/callback.py [hunk 1]
```diff
@@ -230,10 +230,10 @@ def _callback(env):
                 continue  # train data for lgb.cv or sklearn wrapper (underlying lgb.train)
             elif env.iteration - best_iter[i] >= stopping_rounds:
                 if verbose:
-                    _log_info('Early stopping, best iteration is:\n[%d]\t%s' % (
-                        best_iter[i] + 1, '\t'.join([_format_eval_result(x) for x in best_score_list[i]])))
+                    _log_info("Early stopping, best iteration is : \n'
+                        f'[{best_iter[i] + 1}]\t{ '\t'.join([_format_eval_result(x) for x in best_score_list[i]])}")
```
Comment (jameslamb) [inline python-package/lightgbm/callback.py:234]:
```suggestion
                    _log_info("Early stopping, best iteration is: \n"
                                    f"[{best_iter[i] + 1}]\t{'\t'.join([_format_eval_result(x) for x in best_score_list[i]])}")
```

These suggested changes should fix these linter errors from https://github.com/microsoft/LightGBM/pull/4142/checks?check_run_id=2233222418:

```text
./python-package/lightgbm/callback.py:233:65: E203 whitespace before ':'
./python-package/lightgbm/callback.py:234:25: E128 continuation line under-indented for visual indent
```

This should also help to fix some other test failures, caused by the fact that the changes in the current diff were mixing single quotes and double quotes.

**Review by jameslamb — CHANGES_REQUESTED** [review]
I'd really like to help you get this pull request and the others you've opened as part of #4136 merged this week.

Please update this pull request to the latest `master`, and please see the suggestion I just left.

If you don't have time right now to finish this and the other pull requests you've opened for #4136, let me know if you'd prefer that I push directly to your branches. I'd be happy to do that to help with the linting errors (and your name would still be on the commits that are merged).

Thanks again for the time and energy you've put into contributing to `LightGBM`!

**Review by StrikerRUS — APPROVED** [review]
Thank you both!

**Pull request merged**